src/TensorValues/Operations.jl

```diff
@@ -220,8 +246,40 @@ function inner(a::SymFourthOrderTensorValue{D},b::MultiValue{Tuple{D,D}}) where
   inner(a,symmetric_part(b))
 end
 
+# a_i = b_ijk*c_jk
+@generated function inner(a::A, b::B) where {A<:MultiValue{Tuple{D1,D2,D3}},B<:MultiValue{Tuple{D2,D3}}} where {D1,D2,D3}
+  ss = String[]
+  for i in 1:D1
+    s = join([ "a[$i,$j,$k]*b[$j,$k]+" for j in 1:D2 for k in 1:D3])
+    push!(ss,s[1:(end-1)]*", ")
+  end
+  str = join(ss)
+  Meta.parse("VectorValue{$D1}(($str))")
+end
+
 const ⊙ = inner
 
+###############################################################
+# Double Contractions w/ products
+###############################################################
+
+# a_ijpm = b_ijkl*c_klpm
+@generated function ⊡(a::A, b::B) where {A<:SymFourthOrderTensorValue{D},B<:SymFourthOrderTensorValue{D}} where D
```

Comment:
I would give a name to this method, e.g., `double_contraction` and then define the symbol, as we do, e.g., for `dot`.